chore: New package tool

nzakas opened the pull request:
<!--
    Thank you for contributing!

    ESLint adheres to the [OpenJS Foundation Code of Conduct](https://eslint.org/conduct).
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/latest/contribute/pull-requests)
    - Update or create tests
    - If performance-related, include a benchmark
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

#### What is the purpose of this pull request?

Creates a tool that bootstraps a new package in the monorepo.

#### What changes did you make? (Give an overview)

- Added `tools/new-pkg.js`
- Added a package template directory

#### Related Issues

<!-- include tags like "fixes #123" or "refs #123" -->

#### Is there anything you'd like reviewers to focus on?

I set the root `package.json` Node.js version to v22.3 so I could use some of the newer Node.js utilities for this, but let me know if that's a problem.

<!-- markdownlint-disable-file MD004 -->

mdjermanovic commented on tools/new-pkg.js:
```diff
@@ -0,0 +1,227 @@
+const readmePath = "./README.md";
+const readmeContent = readFileSync(readmePath, "utf8");
+const readmeLines = readmeContent.split(/\r?\n/gu);
+const readmeStartIndex = readmeLines.findIndex(line =>
+	line.includes("<!--packages-start-->"),
+);
+const readmeEndIndex = readmeLines.findIndex(line =>
+	line.includes("<!--packages-end-->"),
+);
```
We should add `<!--packages-start-->` and `<!--packages-end-->` to README.md

nzakas replied:
Oh geez, not sure how those changes didn't make it in. Thanks!

mdjermanovic commented on README.md:
```diff
@@ -6,11 +6,15 @@ Monorepo for the rewrite of ESLint.
 - [`@eslint/config-array`](packages/config-array)
 - [`@eslint/object-schema`](packages/object-schema)
 - [`@eslint/migrate-config`](packages/migrate-config)
+  <!--packages-end-->
```
```suggestion

<!--packages-end-->
```

mdjermanovic commented on tools/new-pkg.js:
```diff
@@ -0,0 +1,227 @@
+readmeLines.splice(
+	readmeStartIndex + 1,
+	readmeEndIndex - readmeStartIndex - 1,
+	...newReadmeLines,
```
```suggestion
	"",
	...newReadmeLines,
	"",
```

mdjermanovic reviewed (commented):
To fix the formatting error.

mdjermanovic commented on templates/package/rollup.config.js:
```diff
@@ -0,0 +1,24 @@
+	plugins: [
+		copy({
+			targets: [
+				{ src: "src/types.ts", dest: "dist/cjs", rename: "types.cts" },
```
If the types file will be named `types.cts`, then we should probably add a `build-cts.js` script to update references as in https://github.com/eslint/rewrite/commit/f77ba177d4e4c5d2ed828cfd9a5149df2ccb3a7f.

nzakas replied:
Good point. I can't wait until we only support Node.js 22 and can get rid of CommonJS altogether.

nzakas commented:
I realized there's nothing package-specific about the `build-cts.js` tool, so I moved it to the top level to avoid duplication. I also went through and updated all of the packages with CommonJS to use it as I think we probably have issues there too.

mdjermanovic reviewed (approved):
LGTM, thanks!
